release anouncement for 2.25-RC (#187)
geoserver/geoserver.github.io: _posts/2024-03-04-geoserver-2-25-RC-released.md

*A [preview](https://jodygarnett.github.io/geoserver/) is available, which is being used to help test each page.* Testers include:

* Alexandre Gacon
* Andrea Aime
* Henning Lorenz 
* Geospatial Techno
* Peter Smythe
* Marcus Lingenfelder
* Roar Brænden

Thanks to Jody Garnett (GeoCat) for [this initiative](https://github.com/geoserver/geoserver/wiki/GSIP-221) -- we hope that these changes make the documentation more accessable to our community!

![](/img/posts/2.25/mkdocs-preview.png)

## Security Considerations

This a reminder to update to GeoServer 2.24.2 Release (or GeoServer 2.35.5 Release).

Alongside the upcoming GeoServer 2.25.0 release we will "publicly disclose" a list of Common Vulnerabilities and Exposures that have been addressed previously.
 
* If you are working with a [commercial support provider](/support/) that volunteers with the geoserver-security email list they are already informed.
* If you have updated to GeoServer 2.24.2 Release (or GeoServer 2.23.5 Release) you are already patched.

> I hope you enjoy our team's effort to improve communication. The use of the CVE system allows us to reach a wider audience then reads these blog posts.

Reviewer: ```suggestion
I hope you enjoy our team's effort to improve communication. The use of the CVE system allows us to reach a wider audience than reads these blog posts.
```

---
author: Jody Garnett
date: 2024-03-04
layout: post
title: GeoServer 2.25-RC Release
categories:
- Announcements
tags:
- Release
- Release Candidate
release: release_225
version: 2.25-RC
jira_version: 16901
--- 

GeoServer [2.25-RC](/release/2.25-RC/) release is now available
with downloads
([bin](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/geoserver-2.25-RC-bin.zip/download),
[war](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/geoserver-2.25-RC-war.zip/download),
[windows](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/GeoServer-2.25-RC-winsetup.exe/download)), along with 
[docs](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/geoserver-2.25-RC-htmldoc.zip/download) and
[extensions](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/extensions/).

This is a release candidate intended for public review and feedback.
GeoServer 2.25-RC is made in conjunction with GeoTools 31-RC, and GeoWebCache 1.25-RC. 

Thanks to Jody Garnett (GeoCat) for making this release.

## Release Candiate

> Testing and providing feedback on releases is part of the open-source social contract. The development team (and their employers and customers) are responsible for sharing this great technology with you.

Author: I think I cut and pasted this from a previous RC, but we can revise. I am trying to be more straight forward to our community (which does not have the habit of participation and funding seen in other osgeo projects).

I can express the social contract differently but I wish to have the "social contract" made more clear to our community. 

```
GeoServer follows the classic "release early release often" open-source social contract. The development team is happy to share the new GeoServer series with you. To complete the social contract we rely on your testing and feedback of the release candidate.
```

We have a number of configuration changes to keep in mind if you are updating an existing system:

* The longstanding ``ENTITY_RESOLUTION_ALLOWLIST`` [setting](https://docs.geoserver.org/2.25.x/en/user/production/config.html#external-entities-resolution) has been recommended as a way to control the locations available for external entity resolution when parsing XML documents and requests.

  The default has changed from `*` (allowing any location) to allowing the recommended `www.w3.org`, `schemas.opengis.net`, `www.opengis.net` locations used for OGC Web Services, along with the `inspire.ec.europa.eu/schemas` location used by our friends in Europe.

* The FreeMarker Template HTML Auto-escaping is [now enabled by default](https://docs.geoserver.org/2.25.x/en/user/production/config.html#production-config-freemarker-escaping).

* The [spring security firewall](https://docs.geoserver.org/2.25.x/en/user/production/config.html#production-config-spring-firewall) is now enabled by default.

* A new [configuration setting](https://docs.geoserver.org/2.25.x/en/user/production/config.html#static-web-files) is available to limit content served by `geoserver/www` folder. 

>   While this folder is intended to [serve small web applications](https://docs.geoserver.org/2.25.x/en/user/tutorials/staticfiles.html) it has ended up being useful for a wide range of content.

Author: the actual tutorial page is terrible, and the PR made it worse (more of a warning then a tutorial)

For more information on please see the user guide [Update Instructions](https://docs.geoserver.org/2.25.x/en/user/installation/upgrade.html).

In general we do not like to update defaults, preferring to add notes to [production considerations](https://docs.geoserver.org/2.25.x/en/user/production/config.html) with our recommendations. If you have not checked that page in a while please review.

Thanks to Steve Ikeoka and Jody Garnett for these improvements.

* [GEOS-10438](https://osgeo-org.atlassian.net/browse/GEOS-10438) ENTITY_RESOLUTION_ALLOWLIST property not parsing empty setting
* [GEOS-11221](https://osgeo-org.atlassian.net/browse/GEOS-11221) mkdocs preflight rst fixes
* [GEOS-11289](https://osgeo-org.atlassian.net/browse/GEOS-11289) Enable Spring Security StrictHttpFirewall by default
* [GEOS-11297](https://osgeo-org.atlassian.net/browse/GEOS-11297) Escape WMS GetFeatureInfo HTML output by default
* [GEOS-11300](https://osgeo-org.atlassian.net/browse/GEOS-11300) Centralize access to static web files

> ## MkDocs

Author: The changes are part of this release

> ## MkDocs

Reviewer 2: Where can I find the markdown in the RC tag? All I see is the commit that starts amending the RST.
I'm not trying to belittle the work that's going in there, it's great, but the markdown is not part of this RC.

See project [security policy](https://github.com/geoserver/geoserver/blob/main/SECURITY.md) for more information on how security vulnerabilities are managed. 

## MapML Extension

The MapML extension has received a number of updates and improvements. 

Thanks to 
* [GEOS-11294](https://osgeo-org.atlassian.net/browse/GEOS-11294) MapML Add Support to WMS for Vector Representation
* [GEOS-11154](https://osgeo-org.atlassian.net/browse/GEOS-11154) Improve handling special characters in the MapML HTML Page
* [GEOS-11232](https://osgeo-org.atlassian.net/browse/GEOS-11232) Add Zoom scaled layer templates to MapML
* [GEOS-11277](https://osgeo-org.atlassian.net/browse/GEOS-11277) Have MapML TCRS instances work as actual coordinate reference systems
* [GEOS-11286](https://osgeo-org.atlassian.net/browse/GEOS-11286) MapML HTML backlinks are not workspace aware
* [GEOS-11287](https://osgeo-org.atlassian.net/browse/GEOS-11287) MapML throws unclear exceptions when asked to produce maps in unsupported CRSs

Thanks to Joseph Miller (GeoSolutions) for this work.

> ## Release notes

Author: I can update the template so that release notes are at the end.

categories:
- Announcements
tags:
- Release
- Release Candidate
release: release_225
version: 2.25-RC
jira_version: 16901
--- 

GeoServer [2.25-RC](/release/2.25-RC/) release is now available
with downloads
([bin](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/geoserver-2.25-RC-bin.zip/download),
[war](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/geoserver-2.25-RC-war.zip/download),
[windows](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/GeoServer-2.25-RC-winsetup.exe/download)), along with 
[docs](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/geoserver-2.25-RC-htmldoc.zip/download) and
[extensions](https://sourceforge.net/projects/geoserver/files/GeoServer/2.25-RC/extensions/).

This is a release candidate intended for public review and feedback.
GeoServer 2.25-RC is made in conjunction with GeoTools 31-RC, and GeoWebCache 1.25-RC. 

Thanks to Jody Garnett (GeoCat) for making this release.

## Why share a release candidate?

A sensible question to ask is why a "release candidate" is being produced at all - when we do not recommend running such a thing in production.

> GeoServer also follows a "[release early, release often](https://en.wikipedia.org/wiki/Release_early,_release_often)" approach which is where the project shares releases with so  **you can test and provide feedback**.

Reviewer: ```suggestion
GeoServer also follows a "[release early, release often](https://en.wikipedia.org/wiki/Release_early,_release_often)" approach which is where the project shares releases so  **you can test and provide feedback**.
```

This results in a lovely balance:

> * There are not that many developers of GeoServer, do not have lot of GeoSpatial data to test with.

Reviewer: ```suggestion
* There are not that many developers of GeoServer, do not have lot of geospatial data to test with.
```

However I'm not sure what you have in mind. Are you saying the developers don't have a lot of data?

## Upgrade Notes

We have a number of configuration changes when [updating an existing system](https://docs.geoserver.org/2.25.x/en/user/installation/upgrade.html):

* The longstanding ``ENTITY_RESOLUTION_ALLOWLIST`` [setting](https://docs.geoserver.org/2.25.x/en/user/production/config.html#external-entities-resolution) has been recommended as a way to control the locations available for external entity resolution when parsing XML documents and requests.

  The default has changed from `*` (allowing any location) to allowing the recommended `www.w3.org`, `schemas.opengis.net`, `www.opengis.net` locations used for OGC Web Services, along with the `inspire.ec.europa.eu/schemas` location used by our friends in Europe.

* The FreeMarker Template HTML Auto-escaping is [now enabled by default](https://docs.geoserver.org/2.25.x/en/user/production/config.html#production-config-freemarker-escaping).

* The [spring security firewall](https://docs.geoserver.org/2.25.x/en/user/production/config.html#production-config-spring-firewall) is now enabled by default.

> * A new [configuration setting](https://docs.geoserver.org/2.25.x/en/user/production/config.html#static-web-files) is available to limit content served by `geoserver/www` folder. 

Reviewer: ```suggestion
* A new [configuration setting](https://docs.geoserver.org/2.25.x/en/user/production/config.html#static-web-files) is available to limit content served from the `geoserver/www` folder. 
```

>   If you have not met the ``www`` fodler before it is used to share content, and there is a tutorial [serving static files](https://docs.geoserver.org/2.25.x/en/user/tutorials/staticfiles.html).

Reviewer: ```suggestion
  If you have not met the ``www`` folder before it is used to share content, and there is a tutorial [serving static files](https://docs.geoserver.org/2.25.x/en/user/tutorials/staticfiles.html).
```

![](/img/posts/2.25/mapml-crs.png)

This portion builds on top of the work done months ago to support astronomical CRSs, which allows GeoServer to support multiple CRS authorities.

The MapML preview links are now using the new MapML output format, while the old dedicated REST controller has been removed. This allows for better integration of the MapML format in the GeoServer ecosystem. The MapML viewer has also been updated to the latest version:

![](/img/posts/2.25/mapml-viewer.png)

Thanks to Joseph Miller and Andrea Aime (GeoSolutions) for this work, and Natural Resources Canada for sponsoring it.

## Community Module Updates

Much of the new activity in GeoServer starts as a community module. We'd like to remind you that these modules are
not yet supported, and invite you to join the effort by participating in their development, as well as testing them
and providing feedback.

### Raster attribute Table community module

Developed as part of [GEOS-11175](https://osgeo-org.atlassian.net/browse/GEOS-11175), the Raster Attribute Table community module

> allows to leverage the GDAL Raster Attribute Table (RAT) providing a way to associate attribute information for individual pixel values within the raster, to create styles as well as to provide a richer GetFeatureInfo output.

Reviewer: ```suggestion
uses the GDAL Raster Attribute Table (RAT) to provide a way to associate attribute information for individual pixel values within the raster, to create styles as well as to provide a richer GetFeatureInfo output.
```

![](/img/posts/2.25/rat-ui.png)

![](/img/posts/2.25/rat-map.png)
   
For more information see the [user guide](https://docs.geoserver.org/2.25.x/en/user/community/rat/index.html).

> We'd like to thank Andrea Aime (GeoSolutions) for the development and NOAA for the sponsoring.

Reviewer: ```suggestion
We'd like to thank Andrea Aime (GeoSolutions) for the development and NOAA for sponsoring.
```

(or "... for the sponsorship.")

### Graticules for WMS maps

The graticules community module, developed as part of [GEOS-11216](https://osgeo-org.atlassian.net/browse/GEOS-11216),  provides 
a datastore generating graticules for WMS maps, along with a rendering transformation that can be used to label them.

> The module can be used to draw a graticule in WMS maps, as well as to download them same as part of WFS (or in combination with the WPS download module).

Reviewer: ```suggestion
The module can be used to draw a graticule in WMS maps, as well as to download them as part of WFS (or in combination with the WPS download module).
```

![](/img/posts/2.25/graticule-store.png)

![](/img/posts/2.25/graticules.png)

We'd like to thank Ian Turton for development and GeoSolutions for sponsoring the work. 

### GeoServer monitor Kafka storage
   

> The monitoring Kafka storage module, developed as part of [GEOS-11150](https://osgeo-org.atlassian.net/browse/GEOS-11150), allows to store the requests captured by the [monitoring extension](https://docs.geoserver.org/latest/en/user/extensions/monitoring/index.html) into a Kafka topic.

Reviewer: ```suggestion
The monitoring Kafka storage module, developed as part of [GEOS-11150](https://osgeo-org.atlassian.net/browse/GEOS-11150), allows storing the requests captured by the [monitoring extension](https://docs.geoserver.org/latest/en/user/extensions/monitoring/index.html) into a Kafka topic.
```
